nsxt/resource_nsxt_policy_predefined_gateway_policy_test.go

```diff
@@ -0,0 +1,151 @@
+/* Copyright © 2019 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+// TODO: Rewrite this test based on GW Policy data source when available,
+// use 3.1.0 as baseline for the test and remove testAccOnlyLocalManager
+func TestAccResourceNsxtPolicyPredefinedGatewayPolicy_basic(t *testing.T) {
+	path := "/infra/domains/default/gateway-policies/Policy_Default_Infra"
+	testResourceName := "nsxt_policy_predefined_gateway_policy.test"
+	testGatewayResourceName := "nsxt_policy_tier0_gateway.test"
+	description1 := "test 1"
+	description2 := "test 2"
+	tags := `tag {
+            scope = "color"
+            tag   = "orange"
+        }`
+
+	// NOTE: These tests cannot be parallel, as they modify same default policy
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccOnlyLocalManager(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyPredefinedGatewayPolicyBasic(path, description1, tags),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNsxtPolicyGatewayPolicyExists(testResourceName, defaultDomain),
+					resource.TestCheckResourceAttr(testResourceName, "description", description1),
+					resource.TestCheckResourceAttr(testResourceName, "rule.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "1"),
+					resource.TestCheckResourceAttrSet(testResourceName, "revision"),
+				),
+			},
+			{
+				Config: testAccNsxtPolicyPredefinedGatewayPolicyBasic(path, description2, ""),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNsxtPolicyGatewayPolicyExists(testResourceName, defaultDomain),
+					resource.TestCheckResourceAttr(testResourceName, "description", description2),
+					resource.TestCheckResourceAttr(testResourceName, "rule.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "0"),
+					resource.TestCheckResourceAttrSet(testResourceName, "revision"),
+				),
+			},
+			{
+				Config: testAccNsxtPolicyPredefinedGatewayPolicyPrerequisites(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNsxtPolicyTier0Exists(testGatewayResourceName),
+				),
+			},
+		},
+	})
+}
+
+// NOTE: This test assumes clean environment - no Tier0 routers for NSX < 3.1.0
```

Comment:
nit: use "Tier0 gateway" to align with official term?

Reply:
Done, thanks